daq/runner.py

```diff
@@ -253,6 +256,8 @@ def initialize(self):
         LOGGER.debug('Attaching event channel...')
         self.faucet_events = faucet_event_client.FaucetEventClient(self.config)
         self.faucet_events.connect()
+        # sock fileno will change if the socket is ever reconnected.
```

Comment:
I don't know what to make of this comment. I understand what it's saying, but I don't know why it's here.  If it's true (it will change), then doesn't something need to re-call this function to get the new value?

It doesn't seem like this class variable (faucet_events_fd) is very useful, why does it need to exist at all?  It seems like it's introducing a problem of inconsistent state?

How does the stream monitor (called through self.monitor_stream) pick up the new file descriptor?

Reply:
The fd shouldn’t change since the old fd can still be polled as expected. I don’t know why there’s a new one but I can modify the steam monitors reconnect to overwrite the   fd in callbacks dict so this doesn’t need to be there.

Reply:
I found a much simpler way to handle sock fd changes. PTAL my newest commit.